### ISSUE TITLE
Markdown bullets and code snippets interact strangely

### ISSUE THREAD

**Issue by ianmackenzie** [avh4/elm-format#357]
Consider the following example:

```elm
{-| Increment a number.

  - The given number will be incremented by 1.
  - This function is really simple.


    foo x =
        x + 1

-}
foo : Int -> Int
foo x =
    x + 1
```

Running `elm-format` once removes the extra space between the bullet points and code example:

```elm
{-| Increment a number.

  - The given number will be incremented by 1.
  - This function is really simple.

    foo x =
        x + 1

-}
foo : Int -> Int
foo x =
    x + 1
```

Then, just running `elm-format` again on the same file spaces out the bullets and messes up the code formatting:

```elm
{-| Increment a number.

  - The given number will be incremented by 1.

  - This function is really simple.

    foo x =
    x + 1

-}
foo : Int -> Int
foo x =
    x + 1
```

I'm using `elm-format-0.18 0.6.1-alpha` on Windows 10. I guess the code example is getting folded into the last bullet? The workaround is to add another line of text in between the bullets and code example, in which case everything works normally:

```elm
{-| Increment a number.

  - The given number will be incremented by 1.
  - This function is really simple.

Example:

    foo x =
        x + 1

-}
foo : Int -> Int
foo x =
    x + 1
```

It's probably best practice for documentation comments to just more clearly separate bullets from code blocks, but the non-idempotency of `elm-format` in this case seemed like something to mention.

**Comment by avh4** [comment]
Thanks, good catch!

**Comment by avh4** [comment]
Interestingly, this example is a case where elm-markdown (which uses the marked parser) differs from most other markdown implementations.

https://johnmacfarlane.net/babelmark2/?text=Increment+a+number.%0A%0A++-+The+given+number+will+be+incremented+by+1.%0A++-+This+function+is+really+simple.%0A%0A%0A++++increment+x+%3D%0A++++++++x+%2B+1

**Comment by avh4** [comment]
Fixed in 0f180c1f...f51590b1

**Comment by ianmackenzie** [comment]
Wow, quick work, thanks! For what it's worth, that's the formatting result I initially expected.
